### ISSUE TITLE
Let InputEventStage update playground with HardwareSources

### ISSUE THREAD

**Issue by hhoegelo** [nonlinear-labs-dev/C15#2694]
Currently, the audio-engine sends the position of hardware sources in a brute force way to playground. 
It would be a better design, if the InputEventStage would update the playground in a more granular and event based way.

**Comment by hhoegelo** [comment]
@MatthiasSeeber @schlenkibus

**Comment by MatthiasSeeber** [comment]
Proposal

- refactor the C15Synth `syncExternals` task into two separate tasks:
  - `syncExternalMIDIBridge` task (may stay where it currently is)
  - `syncPlayground` task (probably better placed within InputEventStage)
- refactor InputEventStage
  - `onMIDIHwChanged`: always use `onHWChanged` in any case
  - `onHWChanged`: new location for triggering `syncPlayground` (only when the event was not coming from the UI)

**Comment by KlausStrohhaecker** [comment]
Hhm, the source of the HW-source changes is the Playcontroller and it already sends things event-based and pretty granular.
Currently, playcontroller sends the same stuff twice, one time as TCD-messsage to the audio-engine via direct MIDI over USB, and a second time the same data to the playground with an extra detour via the BBB

**Comment by MatthiasSeeber** [comment]
I agree that there is an opportunity to further streamline our communication process (PlayController --> Synth, BBB) here. But the primary concern of this ticket is, that there is a potentially circular event process involving the PlayGround and Synth. This arises from the MIDI implementation, in which the Synth receives new HW Sources not from the PlayController. (And, changes from the UI can lead to the Synth sending out MIDI)

At the moment, there does not appear to be a reason for involving the PlayController into this process. When, after let's say Pedal 1 chaged via MIDI, the PlayController still should have the internal Pedal position. And when the User then changes Pedal 1 internally, there is one abrupt syncronization of the position from the PlayController and all parties now process the right values.

So, we can think about extending our sanitation process by a separate ticket/task in order to clarify PlayController-->Synth/BBB event flow. Otherwise, I think, we're good.

**Comment by MatthiasSeeber** [comment]
After a few more thoughts and device tests, I can now claim that the situation is - as usual - a tiny bit more complicated:

- for Pedals, Bender, Aftertouch: I still think we're good. An internal movement interrupts eternal movements and vice versa. The PlayController doesn't need to be informed when the User changes a Source via UI or MIDI.
- the Ribbons however cause an issue if their input mode is relative: when a Ribbon was changed via UI or MIDI and then the User tries to make a relative change to the Ribbon, there should not be a jump in the position, but there is. The PlayController needs to be informed about this change, otherwise the relative mode behaves as absolute (which I consider a bug).
